hack/tools/go.mod

```diff
@@ -0,0 +1,27 @@
+module sigs.k8s.io/cluster-api/hack/tools
+
+go 1.12
+
+require (
+	github.com/golang/mock v1.4.3
+	github.com/golangci/golangci-lint v1.17.1
+	gonum.org/v1/netlib v0.0.0-20190331212654-76723241ea4e // indirect
+	k8s.io/code-generator v0.18.2
+	sigs.k8s.io/controller-tools v0.3.0
+	sigs.k8s.io/structured-merge-diff v0.0.0-20190525122527-15d366b2352e // indirect
+	sigs.k8s.io/testing_frameworks v0.1.1
+)
+
+replace (
+	// TODO(vincepri) Remove the following replace directives, needed for golangci-lint until
+	// https://github.com/golangci/golangci-lint/issues/595 is fixed.
+	github.com/go-critic/go-critic v0.0.0-20181204210945-1df300866540 => github.com/go-critic/go-critic v0.0.0-20190526074819-1df300866540
+	github.com/golangci/errcheck v0.0.0-20181003203344-ef45e06d44b6 => github.com/golangci/errcheck v0.0.0-20181223084120-ef45e06d44b6
+	github.com/golangci/go-tools v0.0.0-20180109140146-af6baa5dc196 => github.com/golangci/go-tools v0.0.0-20190318060251-af6baa5dc196
+	github.com/golangci/gofmt v0.0.0-20181105071733-0b8337e80d98 => github.com/golangci/gofmt v0.0.0-20181222123516-0b8337e80d98
+	github.com/golangci/gosec v0.0.0-20180901114220-66fb7fc33547 => github.com/golangci/gosec v0.0.0-20190211064107-66fb7fc33547
+	github.com/golangci/ineffassign v0.0.0-20180808204949-42439a7714cc => github.com/golangci/ineffassign v0.0.0-20190609212857-42439a7714cc
+	github.com/golangci/lint-1 v0.0.0-20180610141402-ee948d087217 => github.com/golangci/lint-1 v0.0.0-20190420132249-ee948d087217
+	github.com/timakin/bodyclose => github.com/golangci/bodyclose v0.0.0-20190714144026-65da19158fa2
+	mvdan.cc/unparam v0.0.0-20190124213536-fbb59629db34 => mvdan.cc/unparam v0.0.0-20190209190245-fbb59629db34
+)
```

Comment:
maybe we could get rid of this now. the issue was resolved afaik. It is just that we did not revisit it on CAPM3 side.

Reply:
This lets us have different versions of dependencies for the tools than for the main code, right? That seems like something we may want to keep.

Reply:
as far as I understood it, it was only there as a workaround to get golangci-lint to work properly. now that issue was solved, we don't need it anymore. CAPI removed it : https://github.com/kubernetes-sigs/cluster-api/blob/master/hack/tools/go.mod

Reply:
Oh, sorry, you were just talking about the replace directive. I thought you meant the whole file. Fixed.